components/structures/TopNav/TopNav.js

```diff
@@ -76,11 +76,11 @@ export default function TopNav({
         ) : null}
         {smallerWidth ? (
           <div className={clsx(styles.contentMenuIcon)}>
-            {rightNavIcon || (
+            {rightNavIcon ?? (contentLeft || contentRight || contentMenu) ? (
```

Comment:
`??` would render `"false"` if `rightNavIcon={false}`. Sticking with react convention, we should not render it, hence use the usual `||`.

Also using only logicals, the condition can be replaced with `rightNavIcon || (contentLeft || contentRight || contentMenu) && <Button .... </Button>`